### PR TITLE
Command Menu

### PR DESCRIPTION
**Pull request by Geevies**
* You can now command commandable animals such as guard dogs or Lt. Columbo by alt-clicking them.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/_onclick/hud/radial.dm [hunk 1]
```diff
@@ -312,4 +312,22 @@ var/global/list/radial_menus = list()
...
+	var/mutable_appearance/MA = mutable_appearance(M.icon, M.icon_state)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(H.head?.flags_inv & HIDEFACE)
+			MA = mutable_appearance(H.head.icon, H.head.icon_state)
+		else if(H.wear_mask?.flags_inv & HIDEFACE)
+			MA = mutable_appearance(H.wear_mask.icon, H.wear_mask.icon_state)
+		else
+			var/obj/item/organ/external/head/head = H.organs_by_name[BP_HEAD]
+			MA = mutable_appearance(head.icon, head.icon_state)
```
Comment (NonQueueingMatt) [inline code/_onclick/hud/radial.dm:325]: I am 99% sure that this does nothing, because heads don't have an icon or state

Comment (Geevies) [reply code/_onclick/hud/radial.dm:325]: it worked in testing before i added the overlay

Comment (NonQueueingMatt) [reply code/_onclick/hud/radial.dm:325]: bruh

File: code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm [hunk 2]
```diff
@@ -17,16 +18,73 @@
 	if(!short_name)
 		short_name = name
 
+/mob/living/simple_animal/hostile/commanded/AltClick(mob/user)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm:21]: this does not check for unconscious user, does it

File: code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm [hunk 3]
```diff
@@ -17,16 +18,73 @@
...
+		var/suffix = ""
+		switch(choice)
+			if("stay")
+				suffix = "here"
+			if("attack")
+				var/list/attack_list = list()
+				for(var/mob/M in view(user.client.view, user))
+					if(M == user || M == src)
+						continue
+					var/radial_button = get_mob_head_mutable_appearance(M)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm:52]: You should cache this, you use it 3 times

File: code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm [hunk 4]
```diff
@@ -17,16 +18,73 @@
...
+					attack_list[M.name] = radial_button
+				suffix = show_radial_menu(user, src, attack_list, radius = 42, tooltips = TRUE)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm:54]: You can use RADIAL_INPUT(user, list) here

File: code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm [hunk 5]
```diff
@@ -17,16 +18,73 @@
...
+				if(!suffix)
+					choice = null
+			if("follow")
+				var/list/attack_list = list()
+				for(var/mob/M in view(user.client.view, user))
+					if(M == src)
+						continue
+					var/image/radial_button = get_mob_head_mutable_appearance(M)
+					attack_list[M.name] = radial_button
+				suffix = show_radial_menu(user, src, attack_list, radius = 42, tooltips = TRUE)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm:64]: Ditto

File: code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm [hunk 6]
```diff
@@ -17,16 +18,73 @@
...
+					var/radial_button = get_mob_head_mutable_appearance(M)
+					attack_list[M.name] = radial_button
+				suffix = show_radial_menu(user, src, attack_list, radius = 42, tooltips = TRUE)
+				if(!suffix)
+					choice = null
+			if("follow")
+				var/list/attack_list = list()
+				for(var/mob/M in view(user.client.view, user))
+					if(M == src)
+						continue
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm:61]: The way you build this list here and in "attack" are basically the same, I'm pretty sure you can unify the two in some way

**Comment by skull132** [comment]
Stale since june. Closing.

**Pull request closed without merging**